src/AudioPlugin/PluginProcessor.cpp

```diff
@@ -135,6 +135,15 @@ void AudioPluginAudioProcessor::processBlock(juce::AudioBuffer<float> &buffer,
   const auto &input = buffer.getWritePointer(0);
   const auto &output = buffer.getWritePointer(1);
 
+  // Send input buffer to editor
+  if (this->editor != nullptr) {
+    this->editor->processAudioBlock(buffer);

```

Comment:
Just for my own understanding, is this explicit delegation how JUCE intends for subcomponents to communicate?

Reply:
I was looking into this and it seemed surprisingly non-obvious how this should be done. Most examples are for `AudioAppComponent`s which seem to hide this completely. I was able to find some indications that this is an acceptable way to go, but I think it is not the only way and I'm very unsure actually.

It _seems_ fine, since `processAudioBlock` doesn't do any heavy uh... processing... it just pushes samples onto a queue, so it's not going to block the audio thread. I also think it looks fine for the `PluginProcessor` to initiate this communication since it is responsible for creating the editor component as well.